src/Button/ButtonBase.tsx

```diff
@@ -1,10 +1,16 @@
+import {WeakValidationMap} from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
-import {COMMON, LAYOUT} from '../constants'
+import {COMMON, LAYOUT, SystemCommonProps, SystemLayoutProps} from '../constants'
 import theme from '../theme'
 import buttonBaseStyles from './ButtonStyles'
 import {compose, variant, fontSize} from 'styled-system'
+import {SxProp} from '../sx'
+import {ComponentProps} from '../utils/types'
 import systemPropTypes from '@styled-system/prop-types'
+import {FontSizeProps} from 'styled-system'
+
+export const systemStyles = compose(fontSize, COMMON, LAYOUT)
```

Comment:
Maybe let's give this a better name:

```suggestion
export const buttonSystemProps = compose(fontSize, COMMON, LAYOUT)
```

Reply:
Since this gets exported from this file instead of being applied to ButtonBase. Maybe we should export a `ButtonSystemProps` type that the other components can use instead of adding `FontSizeProps & SystemCommonProps &
  SystemLayoutProps` to the `StyledButtonBaseProps` type.

Reply:
> Maybe let's give this a better name:

Since `systemStyles` is [the name used in the current published version](https://github.com/primer/components/blob/b894ead466822e42b8f659f37a37ec3c71876506/src/Button/ButtonBase.js#L32), would renaming (which I agree we should do, for the reasons you stated) it be a breaking change?

> Maybe we should export a `ButtonSystemProps` type that the other components can use instead of adding `FontSizeProps & SystemCommonProps & SystemLayoutProps` to the `StyledButtonBaseProps` type.

Great idea—this is an elegant way to address https://github.com/primer/components/pull/1017#discussion_r571254720. 👍

Reply:
Fixed by https://github.com/primer/components/pull/1017/commits/7f8997e9748aaadd949c420b46915396508e8532 and https://github.com/primer/components/pull/1017/commits/80317dac298e3f442aaa535bad19fcf18b808089